src/table/use-column-widths.tsx

```diff
@@ -23,8 +23,8 @@ function readWidths(
   const result = new Map<PropertyKey, number>();
   for (let index = 0; index < visibleColumns.length; index++) {
     const column = visibleColumns[index];
-    let width = (column.width as number) || 0;
-    const minWidth = (column.minWidth as number) || width || DEFAULT_COLUMN_WIDTH;
+    let width = (treatAsNumber(column.width) && (column.width as number)) || 0;
```

Comment:
I don't see this logic using the percentage, am I missing something? If `column.width` is a percentage like `3%`, `treatAsNumber` will return false so `width` will be 0.

Reply:
As it is 0 it then hits the `!width` condition below, and reads the current width from the DOM element. So basically, this part of code 'freezes' the percentage-calculated width

https://github.com/cloudscape-design/components/pull/2553/files#diff-1c6f5a8d52d6e96b2165e4eeb6d8bcbe2674cf66f0e03f078a1e69e8767f5c11R29

Reply:
Is there a chance to make this code easier to follow? We can maybe add a helper function for width and min-width extraction